src/status_im2/contexts/wallet/events.cljs

```diff
@@ -0,0 +1,40 @@
+(ns status-im2.contexts.wallet.events
+  (:require [utils.re-frame :as rf]
+            [taoensso.timbre :as log]
+            [utils.security.core :as security]
+            [native-module.core :as native-module]))
+
+(rf/reg-event-fx 
+:wallet-2/create-derived-addresses
+                 (fn [{:keys [db]} [password {:keys [path]} on-success]]
+                   (let [{:keys [wallet-root-address]} (:profile/profile db)
+                         sha3-pwd   (native-module/sha3 (str (security/safe-unmask-data password)))]
+                     {:json-rpc/call [{:method     "wallet_getDerivedAddresses"
+                                       :params     [sha3-pwd (clj->js wallet-root-address) (clj->js [path])]
+                                       :on-success on-success
+                                       :on-error   #(log/info "failed to derive address " %)}]})))
+(rf/reg-event-fx :wallet-2/add-account
+                 (fn [{:keys [db]} [password {:keys [emoji account-name color]} {:keys [public-key address  path]}]]
+                   (let [key-uid   (get-in db [:profile/profile :key-uid])
+                         sha3-pwd   (native-module/sha3  (security/safe-unmask-data password))
+                         account-config (clj->js {:key-uid key-uid
+                                                  :wallet false
+                                                  :chat false
+                                                  :type :generated
+                                                  :name account-name
+                                                  :emoji emoji
+                                                  :path path
+                                                  :address  address
+                                                  :public-key public-key
+                                                  :colorID color})]
+                     {:json-rpc/call [{:method     "accounts_addAccount"
+                                       :params     [sha3-pwd account-config]
+                                       :on-success #(rf/dispatch [:navigate-to :wallet-accounts])
```

Comment:
FYI: `:on-success` can be an _event vector_ too, leaving the dispatch responsibility to the `:json-rpc/call` effect and being more friendly for unit testing (because of equality checks).

```clojure
:on-success [:navigate-to :wallet-accounts]
```